Pocket/pocket-ios
# Highlights: display existing highlights

## Summary
* This PR lets you display existing highlights in the Reader

## Implementation Details
* Add `ArticleComponentHighlights`, a collection of extensions that enable highlights:
    - extend `[ArticleComponent]` to add the `highlighted(_ patches: [String]) -> [ArticleComponent]` method: this method uses `DiffMatchPatch` to apply the given patches to the components. Components will be transformed in a single `String blob`, then, after the patches are applied, will be deconstructed back into an array of `ArticleComponent`
    - extend `NSAttributedString` to find the tags applied by the above step and convert them into highlighted text

> [!NOTE]
> This PR does not cover adding new highlights. These will be added in a separate PR.

## Test Steps
* Build/run this branch and login with an account that has articles with existing highlights
* Make sure the highlights appear in the reader

## PR Checklist:
